src/Sylius/Component/Payment/Exception/PaymentRequestNotFoundException.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Exception;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/** @experimental */
+final class PaymentRequestNotFoundException extends NotFoundHttpException
+{
+    public function __construct()
+    {
+        parent::__construct('Payment request not found.');
+    }
```

Comment:
It should look like any other exception (message + status code + throwable passed via the constructor).